docs/chart_template_guide/yaml_techniques.md

```diff
@@ -0,0 +1,326 @@
+But when inferring types, the following rules tend to hold true.
+
+If an integer or float is an unquoted bare word, it is typically treated as
+a numeric type:
+
+```yaml
+count: 1
+size: 2.34
+```
+
+But if they are quoted, they are treated as strings:
+
+```yaml
+count: "1" # <-- string, not int
+size: '2.34' # <-- string, not float
+```
+
+The same is true of booleans:
+
+```yaml
+isGood: true   # bool
+answer: "true" # string
+```
+
+The word for an empty value is `null` (not `nil`).
+
+Note that `port: "80"` is valid YAML, and will pass through both the
+template engine and the YAML parser, but will fail if Kubernetes expects
+`port` to be an integer.
+
+In some cases, you can force a particular type inference using YAML node tags:
+
+```yaml
+coffee: "yes, please"
+age: !!str 21
+port: !!int "80"
+```
+
+In the above, `!!str` tells the parser that `age` is a string, even if it looks
+like an int. And `port` is treated as an int, even though it is quoted.
+
+
+## Strings in YAML
+
+Much of the data that we place in YAML documents are strings. YAML has more than
+one way to represent a string. This section explains the ways and demonstrates
+how to use some of them.
+
+There are three "inline" ways of declaring a string:
+
+```yaml
+way1: bare words
+way2: "double-quoted strings"
+way3: 'single-quoted strings'
+```
+
+All inline styles must be on one line.
+
+- Bare words are unquoted, and are not escaped. For this reason, you have to
+  be careful what characters you use.
+- Double-quoted strings can have specific characters escaped with `\`. For
+  example `"\"Hello\", she said"`. You can escape line breaks with `\n`.
+- Single-quoted strings are "literal" strings, and do not use the `\` to
+  escape characters. The only escape sequence is `''`, which is decoded as
+  a single `'`.
+
+In addition to the one-line strings, you can declare multi-line strings:
+
+```yaml
+coffee: |
+  Latte
+  Cappuccino
+  Espresso
+```
+
+The above will treat the value of `coffee` as a single string equivalent to
+`Latte\nCappuccino\nEspresso\n`.
+
+Note that the first line after the `|` must be correctly indented. So we could
+break the example above by doing this:
+
+```yaml
+coffee: |
+         Latte
+  Cappuccino
+  Espresso
+
+```
+
+Because `Latte` is incorrectly indented, we'd get an error like this:
+
+```
+Error parsing file: error converting YAML to JSON: yaml: line 7: did not find expected key
+```
+
+In templates, it is sometimes safer to put a fake "first line" of content in a
+multi-line document just for protection from the above error:
+
+```yaml
+coffee: |
+  # Commented first line
+         Latte
+  Cappuccino
+  Espresso
+
+```
+
+Note that whatever that first line is, it will be preserved in the output of the
+string. So if you are, for example, using this technique to inject a file's contents
+into a ConfigMap, the comment should be of the type expected by whatever is
+reading that entry.
+
+### Controlling Spaces in Multi-line Strings
```

Comment:
It might be useful to add a subsection at the end here about using the `include` or `Files.Get` helper, since you generally need to make sure the template is not indented, and then use the indent function to get the correct level of indentation, such as in https://github.com/kubernetes/charts/blob/master/stable/mariadb/templates/configmap.yaml#L13